### PR TITLE
Fix native auth TTL and PAM prompt, remove options to show password in iinit (main)

### PR DESCRIPTION
**Pull request by alanking**
Companion PR: https://github.com/irods/irods/pull/6391

Still need to run through full test suite, but things were working when tested by hand

### CODE REVIEW

File: src/iinit.cpp [hunk 1]
```diff
@@ -459,4 +435,12 @@ void usageTTL() {
     printf( "administrator (usually a few days).  With the --ttl option, you can\n" );
     printf( "specify how long this derived password will be valid, within the\n" );
     printf( "limits set by the administrator.\n" );
+    printf( "\n" );
+    printf( " -l  list the iRODS environment variables (only)\n" );
+    printf( " -v  verbose\n" );
+    printf( " -V  Very verbose\n" );
+    printf( "--ttl ttl  set the password Time To Live (specified in hours)\n" );
+    printf( "           Run 'iinit -h --ttl' for more\n" );
+    printf( " -h  this help\n" );
```
Comment (korydraughn) [inline src/iinit.cpp:444]: Let's add an `Options:` header just above L439.

Consider aligning the option descriptions. Maybe `--ttl ttl`'s text is moved to the line below and aligned with the others.

Also, `--ttl ttl` -> `--ttl TTL`

File: src/iinit.cpp [hunk 2]
```diff
@@ -431,22 +421,8 @@ int main( int argc, char **argv )
...
     printf( "to be used automatically by the icommands.\n" );
-    printf( "Usage: %s [-ehvVl] [--ttl TimeToLive] [password]\n", prog );
-    printf( " -e  echo the password as you enter it (normally there is no echo)\n" );
-    printf( " -l  list the iRODS environment variables (only)\n" );
-    printf( " -v  verbose\n" );
-    printf( " -V  Very verbose\n" );
-    printf( "--ttl ttl  set the password Time To Live (specified in hours)\n" );
-    printf( "           Run 'iinit -h --ttl' for more\n" );
-    printf( " -h  this help\n" );
+    printf( "Usage: %s [-hvVl] [--ttl TimeToLive]\n", prog );
```
Comment (korydraughn) [inline src/iinit.cpp:424]: Let's capitalize the `--ttl` argument placeholder. That appears to be the convention followed by many Linux commands (e.g. `--ttl TTL`).

**Review by korydraughn — COMMENTED** [review]
Code looks fine to me.

File: src/iinit.cpp (hunk 1 above)
Comment (alanking) [reply src/iinit.cpp:444]: Sounds good

File: src/iinit.cpp (hunk 2 above)
Comment (alanking) [reply src/iinit.cpp:424]: Will do

File: src/iinit.cpp (hunk 1 above)
Comment (alanking) [reply src/iinit.cpp]: Following commit addresses it. If it helps, here's what it looks like:
```
iinit -h
Creates a file containing your iRODS password in a scrambled form,
to be used automatically by the icommands.
Usage: iinit [-hvVl] [--ttl TTL]

When using regular iRODS passwords you can use --ttl (Time To Live)
to request a credential (a temporary password) that will be valid
for only the number of hours you specify (up to a limit set by the
administrator).  This is more secure, as this temporary password
(not your permanent one) will be stored in the obfuscated
credential file (.irodsA) for use by the other iCommands.

When using PAM, iinit always generates a temporary iRODS password
for use by the other iCommands, using a time-limit set by the
administrator (usually a few days).  With the --ttl option, you can
specify how long this derived password will be valid, within the
limits set by the administrator.

Options:
 -l  list the iRODS environment variables (only)
 -v  verbose
 -V  Very verbose
 --ttl TTL
     set the password Time To Live (specified in hours)
 -h  this help

iRODS Version 4.3.0                iinit
```
Not sure if that's what you had in mind

**Comment by trel** [comment]
looking good to me

File: src/iinit.cpp (hunk 1 above)
Comment (korydraughn) [reply src/iinit.cpp]: Yes, that is correct. Just one last thing. Add a newline above the "Usage:" line and I think this PR will be ready.

Comment (alanking) [reply src/iinit.cpp]: Done

**Review by korydraughn — APPROVED** [review]

**Pull request merged**